### PR TITLE
Change order of fields inside Response

### PR DESCRIPTION
**Pull request by IvanGoncharov**
Currently, the spec requires `data` to be the first field in a response.
But after [my PR](https://github.com/graphql/graphql-js/pull/838) was merged into `graphql-js`, `errors` always precede `data`:
image

I still think that having `errors` as the first property is a big DX improvement so I updated the spec to reflect the change in `graphql-js`.

### CODE REVIEW

**Comment by acjay** [comment]
Why should the order matter? I like having `errors` first, but it seems arbitrary. And what of JSON libraries that leave order undefined?

**Comment by jlouis** [comment]
I've always found the ordering requirements odd.

At most, the specification can put a *should* on them, because JSON encoders are free to do whatever they want. If the internal representation of a JSON dict is a large hash table, then the rendering order is going to be arbitrary. Furthermore, some languages (Go is one example) randomizes the order in which you visit a hash table in order to make it explicit to the programmer that they cannot rely on the ordering.

In short, requiring order here is rather brittle and is bound to create problems down the line.

**Comment by IvanGoncharov** [comment]
@acjay @jlouis GraphQL spec only recommends order of properties, for more details see 
https://github.com/facebook/graphql/commit/b22c73c875807e0cbb2c1d51804311e42cba0a36

This PR just makes `graphql-js` (reference implementation) in line with spec recommendation :)

**Comment by stubailo** [comment]
I think order requirements in the spec are for parsers which rely on the order to more efficiently parse the result. IMO that makes errors at the front even more valuable, since it is super useful to know about errors first in the case of a huge result. If it's framed as a recommendation it shouldn't be an issue since things are still valid GraphQL even if they return data first.

**Comment by IvanGoncharov** [comment]
> If it's framed as a recommendation it shouldn't be an issue since things are still valid GraphQL even if they return data first.

You right. But I think the spec should recommend only best practices and error first approach provide better UX. Moreover spec recommendation should be implemented at least in the reference implementation.

**Comment by stubailo** [comment]
Yep, I just think there's a big difference between "recommend" vs. "require" in a spec! I totally agree errors first is the best approach.

**Comment by leebyron** [comment]
"A response to a GraphQL operation must be a map." does not imply order (otherwise it would say "ordered map") - I agree that the "first" and "next" terminology is more ambiguous than helpful in this context, though I like declaring the order via the order of paragraphs.

I think we can add a more explicit phrase about ordering errors first in a formatted output as a convenience

**Comment by IvanGoncharov** [comment]
@leebyron Great improvment 👍 
Hope it will be merged soon.
